app/views/admins/show.html.erb

```diff
@@ -24,21 +24,24 @@
     </div>
   </div>
 
-  <div id="facility-access">
+  <div id="facility-access" data-page="show">
     <div class="card access-tree show">
       <div class="access-tree__section-header">
         Facility access
       </div>
-      <% access_tree = @admin.visible_access_tree %>
-      <%= render access_tree[:render_partial],
-          tree: access_tree[:data],
-          root: access_tree[:root],
-          user_being_edited: nil,
-          tree_depth: 0,
-          page: :show,
-          form: nil %>
+      <%= render_async(access_tree_admin_path(current_admin, page: :show),
+          error_message: '<p class="load-error-message">There was an error, <a href=""> please try again </a></p>') do %>
+        <div class="spinner-wrapper">
+          <div class="spinner spinner-border text-primary" role="status">
+            <span class="sr-only">Loading...</span>
+          </div>
+        </div>
+      <% end %>
     </div>
   </div>
+  <script>
+    new AdminAccess().onAsyncLoaded()
```

Comment:
I'd love to pass the `facility-access` id to the initialization here. `new AdminAccess('#facility-access')`. It decouples the markup specifics from the JS class, and eventually lets us instantiate multiple AdminAccess panels too. Since it's low lift, can we make that change?

Even if there are three or four different handles that need to passed in, I think it's still worth decoupling. What do you think?